### PR TITLE
Mongo Level works on docker and setup page - Dev#373

### LINKED ISSUE #373: NoSQL / MongoDB Level not working after services separated in docker
When accessing the NoSQL level the following error is observed;

```bash
MongoDb Error caught - com.mongodb.MongoTimeoutException: Timed out after 10000 ms while waiting to connect. Client view of cluster state is {type=Unknown, servers=[{address=127.0.0.1:27017, type=Unknown, state=Connecting, exception={com.mongodb.MongoException$Network: Exception opening the socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
```

The mongoSchema.js should be reviewed and possibly a properties file created to connect to mongo.

### PR DESCRIPTION
**Pull request by ismisepaul**
This closes #373 and also adds mongodb set up to the setup page

### CODE REVIEW

File: src/main/java/dbProcs/Database.java [hunk 1]
```diff
@@ -6,27 +6,27 @@
 
 import org.apache.log4j.Logger;
 
-/**
```
Comment (markdenihan) [inline src/main/java/dbProcs/Database.java:9]: Did you make any changes to Database.java? This just looks like whitespace changes

File: src/main/java/servlets/Setup.java [hunk 2]
```diff
@@ -65,16 +72,28 @@ public void doPost (HttpServletRequest request, HttpServletResponse response)  t
...
+			mongoProp.append("\n");
+
 			if (!auth.equals(dbAuth)) {
 				htmlOutput = bundle.getString("generic.text.setup.authentication.failed");
 			} else {
-				Files.write(Paths.get(Constants.DBPROP), dbProp.toString().getBytes(), StandardOpenOption.CREATE);	
-				if (Database.getDatabaseConnection(null) == null) {
+				Files.write(Paths.get(Constants.DBPROP), dbProp.toString().getBytes(), StandardOpenOption.CREATE);
+                Files.write(Paths.get(Constants.MONGO_DB_PROP), mongoProp.toString().getBytes(), StandardOpenOption.CREATE);
+                if (Database.getDatabaseConnection(null) == null ||
```
Comment (markdenihan) [inline src/main/java/servlets/Setup.java]: Formatting here looks a bit off?

File: src/main/java/servlets/Setup.java [hunk 3]
```diff
@@ -65,16 +72,28 @@ public void doPost (HttpServletRequest request, HttpServletResponse response)  t
...
+                        MongoDatabase.getMongoDbConnection(null) == null) {
```
Comment (markdenihan) [inline src/main/java/servlets/Setup.java]: Formatting here looks a bit off?

File: src/main/java/servlets/Setup.java [hunk 4]
```diff
@@ -65,16 +72,28 @@ public void doPost (HttpServletRequest request, HttpServletResponse response)  t
...
 					htmlOutput = bundle.getString("generic.text.setup.connection.failed");
 				} else {
 					try {
 						if (dbOverride.equalsIgnoreCase("overide")) {
 							executeSqlScript();
+                            executeMongoScript();
```
Comment (markdenihan) [inline src/main/java/servlets/Setup.java]: Formatting here looks a bit off?

File: src/main/java/servlets/module/challenge/NoSqlInjection1.java [hunk 5]
```diff
@@ -3,45 +3,42 @@
...
 
-import com.mongodb.BasicDBObject;
-import com.mongodb.DB;
-import com.mongodb.DBObject;
-import com.mongodb.DBCollection;
-import com.mongodb.DBCursor;
-import com.mongodb.MongoClient;
-import com.mongodb.MongoCredential;
-import com.mongodb.MongoException;
+import com.mongodb.*;
```
Comment (markdenihan) [inline src/main/java/servlets/module/challenge/NoSqlInjection1.java]: Isn't it bad practice to import everything when we only need a subset of com.mondodb?

File: src/test/resources/conf/mongo.properties [hunk 6]
```diff
@@ -0,0 +1,3 @@
+connectionHost=localhost
```
Comment (markdenihan) [inline src/test/resources/conf/mongo.properties]: This is added to the test properties, but I can't find any unit tests that use it?

**Review by markdenihan — CHANGES_REQUESTED** [review]
Should there be an option to skip MongoDb in the setup wizard? Right now its a lot of setup for one level. With how this stands the DB cannot be configured unless both are done at once

File: src/test/resources/conf/mongo.properties (hunk 6 above)
Comment (ismisepaul) [reply src/test/resources/conf/mongo.properties]: I've added a task to do this as its outside of the scope of this ticket #454

**Comment by ismisepaul** [comment]
@markdenihan I think I've hit all the points there

File: src/test/resources/conf/mongo.properties [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
+connectionPort=27017
+databaseName=shepherdGames
```
Comment (markdenihan) [inline src/test/resources/conf/mongo.properties:3]: This just needs a JUnit test and it will be good to merge

**Review by markdenihan — CHANGES_REQUESTED** [review]

File: src/main/java/servlets/module/challenge/NoSqlInjection1.java (hunk 5 above)
Comment (ismisepaul) [reply src/main/java/servlets/module/challenge/NoSqlInjection1.java]: Not doing this anymore (Intellij seems to do this automatically)

File: src/test/resources/conf/mongo.properties (hunk 7 above)
Comment (ismisepaul) [reply src/test/resources/conf/mongo.properties:3]: Junits added, may need integration test but needs more investigation for Mongo

**Review by markdenihan — APPROVED** [review]

**Pull request merged**